Framework/src/Aggregator.cxx

```diff
@@ -0,0 +1,120 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+///
+/// \file   Aggregator.cxx
+/// \author Barthelemy von Haller
+///
+
+#include <Configuration/ConfigurationInterface.h>
+#include "QualityControl/Aggregator.h"
+#include "QualityControl/RootClassFactory.h"
+#include "QualityControl/AggregatorInterface.h"
+
+using namespace o2::quality_control::checker;
+using namespace o2::quality_control::core;
+using namespace std;
+
+Aggregator::Aggregator(const std::string& aggregatorName, boost::property_tree::ptree configuration)
+{
+  mAggregatorConfig.name = aggregatorName;
+  mAggregatorConfig.moduleName = configuration.get<std::string>("moduleName", "");
+  mAggregatorConfig.policyType = configuration.get<std::string>("policy", "");
+  mAggregatorConfig.className = configuration.get<std::string>("className", "");
+  mAggregatorConfig.detectorName = configuration.get<std::string>("detectorName", "");
+
+  // Params
+  if (configuration.count("aggregatorParameters")) {
+    for (const auto& [_key, _value] : configuration.get_child("aggregatorParameters")) {
+      mAggregatorConfig.customParameters[_key] = _value.data();
+    }
+  }
+
+  ILOG(Info, Devel) << "Creation of a new aggregator " << aggregatorName << ENDM;
+  for (const auto& [_key, dataSource] : configuration.get_child("dataSource")) { // loop through data sources
+    (void)_key;
+
+    if (auto sourceType = dataSource.get<std::string>("type"); sourceType == "Aggregator" || sourceType == "Check") {
+      auto sourceName = dataSource.get<std::string>("name");
+      ILOG(Info, Devel) << "   Found a source : " << sourceName << ENDM;
+
+      if (dataSource.count("QOs") == 0 || dataSource.get<std::string>("QOs") == "all") {
+        ILOG(Info, Devel) << "      (no QOs specified or specified as `all`)" << ENDM;
+        mAggregatorConfig.allMOs = true;
+        // fixme: this is a dirty fix. Policies should be refactored, so this check won't be needed.
+        if (mAggregatorConfig.policyType != "OnEachSeparately") {
```

Comment:
Anyway, does "OnEachSeparately" actually make any sense in case of aggregating? Maybe we should not mention it any examples and throw if it is chosen for an aggregator?

Reply:
agreed, I had removed it already from the doc.